### ISSUE TITLE
Set cc.bit.z in help_sbc ()

### ISSUE THREAD

**Issue by keelak** [raybellis/usim#5]
If this sequence of instructions is run
LDA #0
ORCC #1 
SBCA #255
then the local int variable t will be assigned 0 - 255 - 1 which is 0xffffff00. The Z condition code bit is set to !t before x is set to t & 0xff. Perhaps Z should be set to the value of !x after x has been assigned the lower 8 bits of t.

In this case usim was built with a 32-bit version of gcc 4.9.2 running on a rather old installation of Linux.

**Comment by raybellis** [comment]
Thanks for the report!   I believe your analysis is correct.

In commit a9514624e60e1c2e6052d5490d8b8e738602539d I've pushed a branch named `fix_z_flag` that includes this fix, does the same for the 16 bit version, and also updates a couple of other instructions where it looked like the same error is feasible.

Please try that and let me know, and if it's all good I'll merge the branch.

**Comment by sorenroug** [comment]
I have a unit test in my Java implementation that does the same check.
Edit: Changed to expect the Z-condition to be set. Code also updated on Github.
```
    /**
     * Test the SBCA instruction.
     * A=0x00 - 0xFF - C=1 becomes 0x00
     */
    @Test
    public void SBCA4() {
        myTestCPU.cc.setN(1);
        myTestCPU.cc.setZ(0);
        myTestCPU.cc.setV(1);
        myTestCPU.cc.setC(1);
        setA(0x00);
        myTestCPU.write(0xB00, 0x82);  // SBCA immediate
        myTestCPU.write(0xB01, 0xFF);
        setPC(0xB00);
        myTestCPU.execute();
        assertEquals(0x00, myTestCPU.a.intValue());
        assertEquals(0, myTestCPU.cc.getN());
        assertEquals(1, myTestCPU.cc.getZ());
        assertEquals(0, myTestCPU.cc.getV());
        assertEquals(1, myTestCPU.cc.getC());
    }
```

**Comment by raybellis** [comment]
> I have a unit test in my Java implementation that does the same check.
> Edit: Changed to expect the Z-condition to be set. Code also updated on Github.
> 
> ```
>     /**
>      * Test the SBCA instruction.
>      * A=0x00 - 0xFF - C=1 becomes 0x00
>      */
>     @Test
>     public void SBCA4() {
>         myTestCPU.cc.setN(1);
>         myTestCPU.cc.setZ(0);
>         myTestCPU.cc.setV(1);
>         myTestCPU.cc.setC(1);
>         setA(0x00);
>         myTestCPU.write(0xB00, 0x82);  // SBCA immediate
>         myTestCPU.write(0xB01, 0xFF);
>         setPC(0xB00);
>         myTestCPU.execute();
>         assertEquals(0x00, myTestCPU.a.intValue());
>         assertEquals(0, myTestCPU.cc.getN());
>         assertEquals(1, myTestCPU.cc.getZ());
>         assertEquals(0, myTestCPU.cc.getV());
>         assertEquals(1, myTestCPU.cc.getC());
>     }
> ```

@sorenroug Ironically this particular bug was a back port from either your C++ v1.0 release or the "BDA" version that came before that.   Unfortunately I can't now find any link to those older BDA patches.     My original 1993 release had a different bug in the SBC instruction.

**Comment by raybellis** [comment]
resolved by a9514624e60e1c2e6052d5490d8b8e738602539d
